pkg/controller/networkpolicy/status_controller.go

```diff
@@ -389,3 +407,27 @@ func (c *networkPolicyControl) UpdateAntreaClusterNetworkPolicyStatus(name strin
 	metrics.AntreaClusterNetworkPolicyStatusUpdates.Inc()
 	return updateErr
 }
+
+// GenerateNetworkPolicyCondition generates conditions based on the given error message.
+// Empty error message generates stale NetworkPolicyCondition of True status.
+// Supports ErrNetworkPolicyAppliedToUnsupportedGroup error.
+func GenerateNetworkPolicyCondition(message string) []crdv1alpha1.NetworkPolicyCondition {
+	var conditions []crdv1alpha1.NetworkPolicyCondition
+	switch message {
+	case "":
+		conditions = append(conditions, crdv1alpha1.NetworkPolicyCondition{
+			Type:               crdv1alpha1.NetworkPolicyConditionRealizable,
+			Status:             v1.ConditionTrue,
+			LastTransitionTime: v1.Now(),
+		})
+	case ErrNetworkPolicyAppliedToUnsupportedGroup.Error():
```

Comment:
I think it's more natural to check the type of the error. Since we only need to add the field to "types.NetworkPolicy", it could be error like "RealizationError", not having to be string?

Reply:
Created an `ErrNetworkPolicyAppliedToUnsupportedGroup` struct that implements `error`.